### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by Sakibs**
Features implemented
- dashboard summary page. 
- dashboard individual company page
- stock hist data now displayed with d3 js chart

Features not added in this
- Toggling watch/unwatch button. I could not do it dynamically. I have an implementation with ajax but that requires a page refresh. The closest I got was the ajax request did go through from the back end but the success response was never captured. Right now user should refresh page or go to another page after watching/unwatching

### CODE REVIEW

File: app/lib/yahoo_finance.rb [hunk 1]
```diff
@@ -40,7 +40,13 @@ def fetch_finance_hist(symbol)
 
       days_past = 365
       hist = YahooFinance::get_historical_quotes_days(symbol, days_past)
-      hist_json = hist.map { |e| { id: e } }.to_json
+      hist_array = Array.new
```
Comment (Sakibs) [inline app/lib/yahoo_finance.rb]: Changed way finance data cached so its easier to process and graph

**Comment by chriskonstad** [comment]
If this works, let's get this merged in ASAP so that we can do more testing.  We can fix the UI button later, before the demonstration.

**Comment by Sakibs** [comment]
Now dashboard works for guest users who are not logged in as well

**Comment by Sakibs** [comment]
I think its ready to merge except the buttons. @chriskonstad @apfong @bmwwoo pliss verify

**Comment by chriskonstad** [comment]
:+1: 

FYI: Before the demonstration, but low priority, we should alphabetize the list of Favorite Companies when looking at a company. Also, we should left justify the dropdown text.  But don't worry about that right now.

**Pull request merged**